topicgenerator/src/main/java/io/dockstore/topicgenerator/client/cli/TopicGeneratorClient.java

```diff
@@ -132,28 +122,36 @@ private void generateTopics(TopicGeneratorConfig topicGeneratorConfig, Iterable<
         final OpenAiService openAiService = new OpenAiService(topicGeneratorConfig.openaiApiKey());
         final String outputFileName = OUTPUT_FILE_PREFIX + "_" + AI_MODEL + "_" + Instant.now().truncatedTo(ChronoUnit.SECONDS).toString().replace("-", "").replace(":", "") + ".csv";
 
-        try (CSVPrinter csvPrinter = new CSVPrinter(new FileWriter(outputFileName, StandardCharsets.UTF_8), CSVFormat.DEFAULT.builder().setHeader(OUTPUT_CSV_HEADERS).build())) {
+        try (CSVPrinter csvPrinter = new CSVPrinter(new FileWriter(outputFileName, StandardCharsets.UTF_8), CSVFormat.DEFAULT.builder().setHeader(OutputCsvHeaders.class).build())) {
             for (CSVRecord entry: entriesCsvRecords) {
-                final String trsId = entry.get(TRS_ID_CSV_HEADER);
-                final String versionId = entry.get(VERSION_CSV_HEADER);
+                final String trsId = entry.get(InputCsvHeaders.trsId);
+                final String versionId = entry.get(InputCsvHeaders.version);
 
                 // Get descriptor file content and entry type
                 FileWrapper descriptorFile;
                 String entryType;
+                DescriptorTypeEnum descriptorType;
                 try {
                     entryType = ga4Ghv20Api.toolsIdGet(trsId).getToolclass().getName().toLowerCase();
                     final ToolVersion version = ga4Ghv20Api.toolsIdVersionsVersionIdGet(trsId, versionId);
-                    final String descriptorType = version.getDescriptorType().get(0).toString();
-                    descriptorFile = ga4Ghv20Api.toolsIdVersionsVersionIdTypeDescriptorGet(trsId, descriptorType, versionId);
+                    descriptorType = version.getDescriptorType().get(0);
+                    descriptorFile = ga4Ghv20Api.toolsIdVersionsVersionIdTypeDescriptorGet(trsId, descriptorType.toString(), versionId);
+
+                    if (descriptorType == DescriptorTypeEnum.NFL) {
+                        // For nextflow workflows, find the main script. Otherwise, use the nextflow.config file (which is a nextflow workflow's primary descriptor in Dockstore terms)
+                        Optional<FileWrapper> nextflowMainScript = getNextflowMainScript(descriptorFile.getContent(), ga4Ghv20Api, trsId, versionId, descriptorType);
```

Comment:
Get the main script file if possible. If not, use the nextflow.config to generate the AI topic since a topic is better than no topic